src/test/tests.ts

```diff
@@ -272,6 +272,65 @@ describe('RedisPubSub', () => {
     });
   });
 
+  it('refuses custom reviver with a deserializer', done => {
+    const reviver = stub();
+    const deserializer = stub();
+
+    try {
+      expect(() => new RedisPubSub({...mockOptions, reviver, deserializer}))
+          .to.throw("Reviver and deserializer can't be used together");
+      done();
+    } catch (e) {
+      done(e);
+    }
+  });
+
+  it('allows to use a custom serializer', done => {
+    const serializer = stub();
+    const serializedPayload = `{ "hello": "custom" }`;
+    serializer.returnWith(serializedPayload);
+
+    const pubSub = new RedisPubSub({...mockOptions, serializer });
+
+    try {
+      pubSub.subscribe('TOPIC', message => {
+        try {
+          expect(message).to.eql({hello: 'custom'});
+          done();
+        } catch (e) {
+          done(e);
+        }
+      }).then(() => {
+        pubSub.publish('TOPIC', {hello: 'world'});
+      });
+    } catch (e) {
+      done(e);
+    }
+  });
+
+  it('allows to use a custom deserializer', done => {
```

Comment:
@quentinus95 Please add tests for when serializers or deserializers throws an error

Reply:
What would you expect from such a situation? Current behavior, if I'm not wrong, is that the error would be thrown directly to the subscriber / dispatcher. Do you feel this behavior is wrong?

Reply:
For the deserializer the code just ignores the error and gives back the message unmarshalled:
https://github.com/davidyaha/graphql-redis-subscriptions/pull/199/files#diff-3b3bf6f5369f165d6124779b61832724R162-R166
I feel like that's a behaviour that we can change if such a request arises from the community.
I would much rather have a discussion about that in a dedicated issue rather than change the behaviour.
So for the moment, don't change the package code but regardless, we should have tests demonstrating the behaviour.

Thanks!